apps/prairielearn/src/pages/partials/questionAwardedPoints.ejs

```diff
@@ -1,11 +1,28 @@
 <% include('pointsFormatter'); %>
 
-<% for (var i = 0; i < awarded_points.length; i++) { %>
-    <% if (awarded_points[i] < init_points) { %>
-        <% if (awarded_points[i] > 0) { %>
-            <span class="badge badge-warning"><%= getStringFromFloat(awarded_points[i]) %></span>
+<% const MAX_DISPLAYED_VARIANTS = 10; %>
+
+<% if (previous_variants) { %>
+    <%
+        const collapseId = `variants-points-collapse-${instance_question_id}`;
+        const collapseButtonId = `variants-points-collapse-button-${instance_question_id}`;
+        let startHiddenIndex = null;
+        let endHiddenIndex = null;
+    %>
+    <% if (previous_variants.length > MAX_DISPLAYED_VARIANTS) { %>
+        <a id="<%= collapseButtonId %>" role="button" onclick="$(`#<%= collapseId %>`).css('display','inline'); $(`#<%= collapseButtonId %>`).css('display','none');">&hellip;</a>
```

Comment:
Nit: could this logic be moved to a separate script tag for cleanness? And in that case, either switch to vanilla JS, or use JQuery show/hide.

Reply:
Also, use `button class="btn"` instead of `a role="button"`.

Reply:
I replaced the jQuery with plain JS in 964127603c9aec23044d8e583b6105d91383b545. I left the code in the `onclick` handler because it's quite short and self-contained. In the future we can move this to an asset when we convert this all to `html` templates.

Regarding the use of a button, I actually deliberately chose to style this as a link instead of a button because a button takes up too much vertical space, expanding the row height (and a `btn-xs` uses a too-small font).

Reply:
What about btn-link? https://getbootstrap.com/docs/4.6/components/buttons/#examples

Reply:
That still includes all the usual button padding and margin. I think this is a case where using an actual `<a>` link but indicating that it should function as a button (`role="button"`) is cleanest?

Reply:
Even using `btn btn-link p-0 m-0 border-0 rounded-0` I still couldn't get the button to preserve the line height. I think the button element itself is slightly too high (and using `btn-xs` shrinks the font). I thus left it as a `<a role="button">`.

Reply:
The ellipsis don't have any visual indication that they can be clicked with the a styling, that looks weird to me. I tried with `badge-light`, it looks a little bit better, but I'll leave that decision to you. And I also wasn't able to get the button to style in a decent way.

One thing I tried that made it look a bit better was to replace `&ellip;` with `ctdot;`. Up to you.

Reply:
I switched to a `<button>` by avoiding Bootstrap's `.btn` styling and explicitly styling a built-in button. This gives the correct spacing.

I agree with you that I'd prefer a better visual indication that this element is clickable. It does have the mouse pointer change on hover, which helps for platforms with pointers but not for touch platforms. I tried using a `.badge` but that doesn't play well with `<button>` and I'd rather not fall back to `<a>` now that I've got `<button>` working (for better accessibility support).

I like the idea of `&ctdot;` -- I switched to that.